client/me/purchases/remove-purchase/index.jsx

```diff
@@ -475,15 +486,21 @@ class RemovePurchase extends Component {
 }
 
 export default connect(
-	( state, { selectedSite } ) => ( {
-		isDomainOnlySite: selectedSite && isDomainOnly( state, selectedSite.ID ),
-		isAutomatedTransferSite: selectedSite && isSiteAutomatedTransfer( state, selectedSite.ID ),
-		isChatAvailable: isHappychatAvailable( state ),
-		isChatActive: hasActiveHappychatSession( state ),
-		purchasesError: getPurchasesError( state ),
-		precancellationChatAvailable: isPrecancellationChatAvailable( state ),
-		userId: getCurrentUserId( state ),
-	} ),
+	( state, { purchase, selectedSite } ) => {
+		const isJetpack = purchase && isJetpackPlan( purchase );
+		return {
+			isDomainOnlySite: purchase && isDomainOnly( state, purchase.siteId ),
+			/* We know Jetpack sites are not Atomic */
+			isAutomatedTransferSite:
+				! isJetpack && ( selectedSite ? isSiteAutomatedTransfer( state, selectedSite.ID ) : null ),
```

Comment:
We're using `selectedSite.ID` and `purchase.siteId` inconsistently in this `mapStateToProps`. Can we/Should we get rid of `selectedSite` altogether and only use `purchase.siteId`?

Reply:
I've just reviewed the implementation here and it seems to nicely return `null`, so much of this ceremony is not required.

https://github.com/Automattic/wp-calypso/blob/aa43630d6ef40f25cb8d78c26d3f7b828c89ff3f/client/state/selectors/is-site-automated-transfer.js#L15-L17

The `isJetpack` site can return a `false` early, but I don't think that's adding much value.